### PR TITLE
Mock bazel run in Gazelle integration tests

### PR DESCRIPTION
**Pull request by mai93**
Co-authored-by: Borja Lorente <blorente@apple.com>

This is a simplified version of https://github.com/bazelbuild/intellij/pull/4127 that uses `registerApplicationService()` to register the mock `GazelleRunner` instead of using different plugin xml for tests.

### CODE REVIEW

File: gazelle/src/com/google/idea/blaze/gazelle/GazelleRunner.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * The results may be cached in the workspace.
+ */
+public abstract class GazelleRunner {
+
+    public static GazelleRunner getInstance() {
+        return ApplicationManager.getApplication().getService(
+                GazelleRunner.class);
+    }
+
+    protected static BlazeCommand createGazelleRunCommand(
```
Comment (blorente) [inline gazelle/src/com/google/idea/blaze/gazelle/GazelleRunner.java:44]: If we extract this function, we should use it in `GazelleRunnerImpl.runBlazeGazelle`.

**Review by blorente — APPROVED** [review]
One Small nit, otherwise looks good to me! Thanks for getting it over the line.

File: gazelle/src/com/google/idea/blaze/gazelle/GazelleRunner.java (hunk 1 above)
Comment (mai93) [reply gazelle/src/com/google/idea/blaze/gazelle/GazelleRunner.java:44]: I missed that, thank you!

**Comment by mai93** [comment]
@tpasternak Can you please review this PR? we are still missing an approval to merge, probably because @blorente and I are co-authoring.

**Pull request merged**